speechbrain/utils/evaluate_dialogue_state_tracking.py

```diff
@@ -0,0 +1,685 @@
+    "train-departure",
+    "train-destination",
+]
+
+
+def dialogueState_str2dict(
+    dialogue_state: str, slot_type_filtering: list[str] = None
+):
+    """
+    Converts the ; separated Dialogue State linearization to a domain-slot-value dictionary.
+    When *slot_type_filtering* is provided, it filters out the slots which are not part of this list.
+    """
+    dict_state = {}
+
+    # We consider every word after "[State] " to discard the transcription if present.
+    dialogue_state = dialogue_state.split("[State] ")[-1]
+    if ";" not in dialogue_state:
+        return {}
+    else:
+        slots = dialogue_state.split(";")
+        for slot_value in slots:
+            if "=" not in slot_value:
+                continue
+            else:
+                slot, value = (
+                    slot_value.split("=")[0].strip(),
+                    slot_value.split("=")[1].strip(),
+                )
+                if slot_type_filtering and slot not in slot_type_filtering:
+                    continue
+                elif "-" in slot:
+                    domain, slot_type = (
+                        slot.split("-")[0].strip(),
+                        slot.split("-")[1].strip(),
+                    )
+                    if domain not in dict_state.keys():
+                        dict_state[domain] = {}
+                    dict_state[domain][slot_type] = value
+
+        return dict_state
+
+
+def dialogueState_dict2str(dialogue_state: dict):
+    """
+    Converts a dialogue state as a dictionary per slot type per domain to a serialized ; separated list
+    """
+    slots = []
+    for domain, slots_values in dialogue_state.items():
+        slots.extend(
+            [f"{domain}-{slot}={value}" for slot, value in slots_values.items()]
+        )
+
+    return "; ".join(slots)
+
+
+class JGATracker:
+    """
+    Class to track the Joint-Goal Accuracy during training. Keeps track of the number of correct and total dialogue states.
+    """
+
+    def __init__(self):
+        self.correct = 0
+        self.total = 0
+
+    def append(self, predictions: list[str], targets: list[str]):
+        """
+        This function is for updating the stats according to the a batch of predictions and targets.
+
+        Arguments
+        ----------
+        predictions : list[str]
+            Predicted dialogue states.
+        targets : list[str]
+            Target dialogue states.
+        """
+        for prediction, reference in zip(predictions, targets):
+            pred = dialogueState_str2dict(prediction)
+            ref = dialogueState_str2dict(reference)
+            if pred == ref:
+                self.correct += 1
+            self.total += 1
+
+    def summarize(self):
+        """
+        Averages the current Joint-Goal Accuracy (JGA).
+        """
+        return (
+            round(100 * self.correct / self.total, 2)
+            if self.total != 0
+            else 100.00
+        )
+
+
+class DSTMetrics:
```

Comment:
IMO should be moved to `metric_stats.py` without any dataset-specific things (including defaults) and made a child of `MetricStats`.